fireapp/lib/presentation/constraint_form/constraint_form_view_model.dart

```diff
@@ -59,25 +60,33 @@ class SchedulerConstraintFormViewModel
     _submissionState.add(RequestState.loading());
     () async {
       try {
-        var request = await _schedulerConstraintFormRepository.makeNewRequest(NewRequest(
-          title: titleController.text,
-          status:"",
-        ));
-        await _schedulerConstraintFormRepository.makeVehicleRequest(VehicleRequest(
-            requestId: request.id,
-            startDate: _selectedDate.value!.withTime(_selectedStartTime.value!),
-            endDate: _selectedDate.value!.withTime(_selectedEndTime.value!),
-            assetType: selectedAsset!.code
-        ));
+        String title = titleController.text;
+        int selectedVehicle = mapVehicleTypeToNumber(selectedAsset!.code);
+        var startDate = _selectedDate.value!.withTime(_selectedStartTime.value!);
+        var endDate = _selectedDate.value!.withTime(_selectedEndTime.value!);
+        NewRequestResponse response = await _schedulerConstraintFormRepository.makeNewShiftRequest(title, selectedVehicle, startDate, endDate);
         _submissionState.add(RequestState.success(null));
-        _navigate.add(ConstraintFormNavigation.shiftRequest(request.id));
+        _navigate.add(ConstraintFormNavigation.shiftRequest(response.id.toString()));
       } catch(e, stacktrace) {
         print(stacktrace);
         logger.e(e, stackTrace: stacktrace);
         _submissionState.add(RequestState.exception(e));
       }
     }();
   }
+  
+  int mapVehicleTypeToNumber(String vehicleType) {
```

Comment:
Just pull the id out of the asset

Reply:
The backend required me to encode the vehicle number, where small is 1, medium is 2, and heavy is 3, etc. It’s more than just an ID. I tried to find a way to avoid hardcoding it, but I couldn't come up with one.

Reply:
Yeah some future work will need to be done to un-hardcode it. I would like it if you just pretended they were IDs, because that is what they would essentially be. Take a look at the comment I made on your asset type changes and you should be able to get rid of this function.